cli/cmd/root.go

```diff
@@ -136,86 +123,14 @@ func init() {
 	RootCmd.AddCommand(jaeger.NewCmdJaeger())
 	RootCmd.AddCommand(multicluster.NewCmdMulticluster())
 	RootCmd.AddCommand(viz.NewCmdViz())
-}
-
-type statOptionsBase struct {
-	namespace    string
-	timeWindow   string
-	outputFormat string
-}
-
-func newStatOptionsBase() *statOptionsBase {
-	return &statOptionsBase{
-		namespace:    defaultNamespace,
-		timeWindow:   "1m",
-		outputFormat: tableOutput,
-	}
-}
-
-func (o *statOptionsBase) validateOutputFormat() error {
-	switch o.outputFormat {
-	case tableOutput, jsonOutput, wideOutput:
-		return nil
-	default:
-		return fmt.Errorf("--output currently only supports %s, %s and %s", tableOutput, jsonOutput, wideOutput)
-	}
-}
-
-func renderStats(buffer bytes.Buffer, options *statOptionsBase) string {
-	var out string
-	switch options.outputFormat {
-	case jsonOutput:
-		out = buffer.String()
-	default:
-		// strip left padding on the first column
-		b := buffer.Bytes()
-		if len(b) > padding {
-			out = string(b[padding:])
-		}
-		out = strings.Replace(out, "\n"+strings.Repeat(" ", padding), "\n", -1)
-	}
-
-	return out
-}
-
-// getRequestRate calculates request rate from Public API BasicStats.
-func getRequestRate(success, failure uint64, timeWindow string) float64 {
-	windowLength, err := time.ParseDuration(timeWindow)
-	if err != nil {
-		log.Error(err.Error())
-		return 0.0
-	}
-	return float64(success+failure) / windowLength.Seconds()
-}
-
-// getSuccessRate calculates success rate from Public API BasicStats.
-func getSuccessRate(success, failure uint64) float64 {
-	if success+failure == 0 {
-		return 0.0
-	}
-	return float64(success) / float64(success+failure)
-}
-
-// getDefaultNamespace fetches the default namespace
-// used in the current KubeConfig context
-func getDefaultNamespace() string {
-	rules := clientcmd.NewDefaultClientConfigLoadingRules()
-
-	if kubeconfigPath != "" {
-		rules.ExplicitPath = kubeconfigPath
-	}
-
-	overrides := &clientcmd.ConfigOverrides{CurrentContext: kubeContext}
-	kubeCfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides)
-	ns, _, err := kubeCfg.Namespace()
-
-	if err != nil {
-		log.Warnf(`could not set namespace from kubectl context, using 'default' namespace: %s
-		 ensure the KUBECONFIG path %s is valid`, err, kubeconfigPath)
-		return corev1.NamespaceDefault
-	}
 
-	return ns
+	// Viz Extension sub commands
+	RootCmd.AddCommand(viz.NewCmdDashboard())
+	RootCmd.AddCommand(viz.NewCmdEdges())
+	RootCmd.AddCommand(viz.NewCmdRoutes())
+	RootCmd.AddCommand(viz.NewCmdStat())
+	RootCmd.AddCommand(viz.NewCmdTap())
+	RootCmd.AddCommand(viz.NewCmdTop())
```

Comment:
In order to mark these subcommands as deprecated, you can do something like this for each one: 
```go
cmdRoutes := viz.NewCmdRoutes()
cmdRoutes.Deprecated = "use instead 'linkerd viz routes'"
RootCmd.AddCommand(cmdRoutes)
```

Reply:
Created https://github.com/linkerd/linkerd2/issues/5523